### DISCUSSION TITLE
PMD Exception while using insert "as user"

### DISCUSSION THREAD

**Discussion by vasantha77-surya** [pmd/pmd#4734] [Q&A] [answered]
To Upload our app in app exchange we are doing a security review through the salesforce cli scanner. after the execution of the scanner we got the results in csv format there were a couple of PMD errors and exceptions. The PMD error was that when performing any DML operation we were supposed to do it as insert as user or update as user. On doing this the PMD error was rectified but the PMD exception was obtained in the result csv file.
I am not sure why I am getting this exception. can you help me how to overcome this scenario?

**Comment by adangel** [comment, marked as answer]
This new Apex syntax is only supported with the latest PMD version 7.0.0-rc4 (see https://docs.pmd-code.org/latest/pmd_release_notes.html#apex-jorje-updated or issue #3973)

Please make sure, that salesforce cli scanner uses the latest PMD version.

**Reply by vasantha77-surya** [reply]
Can you please give me the detailed steps or any documentation to refer to update the PMD version currently my PMD version is 0.6.2 and the API version of all the apex classes is 57 I have updated my cli as well
